solr/solrj/src/java/org/apache/solr/client/solrj/io/stream/JDBCStream.java

```diff
@@ -283,8 +283,9 @@ public void open() throws IOException {
       resultSet = statement.executeQuery(sqlQuery);
       resultSet.setFetchSize(fetchSize);
     } catch (SQLException e) {
-      throw new IOException(String.format(Locale.ROOT, "Failed to execute sqlQuery '%s' against JDBC connection '%s'.\n"
-          + e.getMessage(), sqlQuery, connectionUrl), e);
+      // don't embed the exception message in the format template as wildcard '%' will throw off the formatter
```

Comment:
I'm confused here, why does exception message have a %, but sqlQuery wouldn't?

Reply:
No actually you're right, if the SQL has a %, then it will confuse the formatter too. This was just something I found while debugging another issue and it looked like it was the exception message that caused the issue but the sql query would too. Will escape any percents in the query / error message with %% and add a test case

Reply:
Wait, no ... now I'm confusing myself ;-) The reason the exception message and not the sqlQuery was messing up the formatter is the code (before I changed it) was appending the exception message to the formatter template whereas the sqlQuery is injected as a %s parameter. I think the better fix here is to just inject the exception message as a %s parameter as well